src/data/number_types.jl

```diff
@@ -3,20 +3,19 @@
 # representations than ordinary opaque types
 
 # This construction prevents these methods from getting called on type unions
-const SignedTypes        = Union{Type{Int8}, Type{Int16}, Type{Int32}, Type{Int64}, Type{Int128}}
-const UnsignedTypes      = Union{Type{UInt8}, Type{UInt16}, Type{UInt32}, Type{UInt64}, Type{UInt128}}
-const FloatTypes         = Union{Type{Float16}, Type{Float32}, Type{Float64}}
-const PrimitiveTypeTypes = Union{SignedTypes, UnsignedTypes, FloatTypes, Type{Bool}}
-const PrimitiveTypes     = Union{Bool, Int8, Int16, Int32, Int64, Int128, UInt8, UInt16, UInt32,
-                                 UInt64, UInt128, Float16, Float32, Float64}
+const SignedTypes        = [Int8, Int16, Int32, Int64, Int128]
+const UnsignedTypes      = [UInt8, UInt16, UInt32, UInt64, UInt128]
+const FloatTypes         = [Float16, Float32, Float64]
+const PrimitiveTypes     = [SignedTypes; UnsignedTypes; FloatTypes; Bool]
+const PrimitiveTypeTypes = Union{(Type{T} for T in PrimitiveTypes)...}
```

Comment:
Is there a specific reason for this change?
It's not a problem, I'm just curious. I've never touched this code myself.

Reply:
This simplifies the code below. Before, it created a `Union` (instead of an array), and then use an internal method of `Base` to iterate over the types in that `Union`, and accessed the size of these types. The code is now simpler and doesn't use any undocumented functions any more. (One of these undocumented function was accessing the size of a type in a way that is not supported in Julia 1.9 any more.)

Reply:
thank you